openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -2166,6 +2191,96 @@ The following is a non-normative example of a Credential Response containing a C
+
+Since the key attestations may have large audience as many Credential Issuers that not necessarily uses the same trust framework or internal design decisions, it is required to use a common approach to facilitate interoperability. Therefore, key attestations SHOULD use a common format, allowing Credential Issuers to develop consistent evaluation processes, reducing complexity and potential errors. Common formats makes easy for Credential Issuers to demonstrate compliance with regulatory requirements across different jurisdictions, they also facilitate the development of shared best practices and security benchmarks.
+
+There are two flows how key attestations can be used within Credential issuance:
+- The Wallet uses the `jwt` proof type in the Credential Request to create a proof of possession of the key and adds the key attestation in the JOSE header as specified in (#jwt-proof-type).
+- The Wallet uses the `attestation` proof type in the Credential Request with the key attestation without a proof of possession of the key itself as specified in (#attestation-proof-type).
+
+The latter may avoid unnecessary user interaction during the Credential issuance, as the key itself is not performing a signature operation.
+
+## Key Attestation in JWT format {#keyattestation-jwt}
+
+The JWT is signed by the Wallet Provider or the Wallet's key storage component itself and contains the following elements:
+
+* in the JOSE header,
+  * `alg`: REQUIRED. A digital signature algorithm identifier such as per IANA "JSON Web Signature and Encryption Algorithms" registry [@IANA.JOSE.ALGS]. It MUST NOT be `none` or an identifier for a symmetric algorithm (MAC).
+  * `typ`: REQUIRED. MUST be `keyattestation+jwt`, which explicitly types the key proof JWT as recommended in Section 3.11 of [@!RFC8725].
+
+The key attestation may use `x5c`, `kid` or `trust_chain` (as defined in (#jwt-proof-type) ) to convey the public key and the associated trust mechanism to sign the key attestation.
+
+* in the JWT body,
+  * `iat`: REQUIRED (number). Integer for the time at which the key attestation was issued using the syntax defined in [@!RFC7519].
+  * `exp`: REQUIRED (number). Integer for the time at which the key attestation and the key(s) it is attesting expire, using the syntax defined in [@!RFC7519].
+  * `attested_keys` : REQUIRED. Array of attested keys from the same key storage component using the syntax of JWK as defined in [@!RFC7517].
```

Comment:
Wondering it this could be an JWKS (https://www.rfc-editor.org/rfc/rfc7517.html#section-5) instead of directly an array if JWK. Having an object here is an extra indirection however it is a standard representation and provides "extra space" to add more common properties to this set of keys.

Reply:
Using JWKS would also allow a single top-level claims, with the extra properties being defined inside the JWKS (which allows for extensibility):
```json
"attested_keys" {
    "keys": [{"...":"..."}, {"...":"..."}],
    "key_type": "strong_box",
    "user_authentication": ["system_biometry", "system_pin"]
}
```

Reply:
Thinking a bit more, a better design could be
```json
"attested_keys": {
    "keys": [{"...":"..."}, {"...":"..."}],
    "attestation": {
        "key_type": "strong_box",
        "user_authentication": ["system_biometry", "system_pin"],
        "a_future_extension": "with some value"
    }
}
```

- A *single* JWT claim registration - `attested_keys`
- A *single* JSON Web Key Set Parameter registeration - `attestation`.
- An extensible object with attestation information, including `key_type`, `user_authentication` and other future properties.

Reply:
Shouldn't this model also allows for key derivation scenarios (e.g. Hierarchical 
Deterministic Keys), where instead of explicitly listing all the keys, the attestation contains some seed parameters for the key generation?

Reply:
I like the proposal. JWKS make sense

Reply:
If attested_keys is a JWKS, I'd prefer:
```json
"attested_keys": {
    "keys": [{"...":"..."}, {"...":"..."}],
    "key_type": "strong_box",
    "user_authentication": ["system_biometry", "system_pin"],
    "a_future_extension": "with some value"
}
```

JWKS Parameters is an existing IANA Registry.

Reply:
My reasoning is that this spec would contribute with a single parameter to the JWKS Parameters IANA register (e.g. `attestation`), which would be an object. The schema of this object and its extensibility would be own by this spec.

Reply:
Brian expresses concerns about registering these parameters to IANA

Reply:
IMO, if `attested_keys` is defined to be a JWKS object as defined in RFC 7517, then it is probably necessary to register included parameters with IANA. Currently, there is only one parameter registered in the IANA JWKS registry which is `keys`. To avoid name clashes, these additional ones have to be registered there as well.

On the other hand, if we would just not define `attested_keys` as a JWKS but instead fully define it which would also require defining `keys`, then no IANA registration is needed.

Reply:
I think we should first figure out which of these additional attributes we want to keep and then revisit this discussion

Reply:
I tend to leave this as top-level claims, but I would leave it to working group decision

Reply:
discussed in a WG call and reached a rough consensus that we will start with the current structure of array of jwks complying objects.